pytripgui/empty_patient_vc/empty_patient_cont.py

```diff
@@ -0,0 +1,446 @@
+        enable_validation_list(validator, [
+            dim[0]["width"],
+            dim[0]["height"],
+        ])
+        dim[0]["pixel_size"].emit_on_text_change(lambda: validate_list([dim[0]["width"], dim[0]["height"]]))
+
+        validator = MultipleOfRegularExpressionValidator(Regex.FLOAT_POSITIVE.value)
+        validator.set_multiple_of(dim[0]["slice_distance"])
+        enable_validation_list(validator, [dim[0]["depth"]])
+        dim[0]["slice_distance"].emit_on_text_change(lambda: validate_list([dim[0]["depth"]]))
+
+        validator = PixelSizeValidator(Regex.FLOAT_POSITIVE.value)
+        validator.set_additional_validation(self._validate_pixel_size)
+        enable_validation_list(validator, [dim[1]["width"], dim[1]["height"]])
+
+        validator = PixelSizeValidator(Regex.INT_POSITIVE.value)
+        validator.set_additional_validation(self._validate_pixel_size)
+        enable_validation_list(validator, [dim[1]["pixel_number_x"], dim[1]["pixel_number_y"]])
+
+    def _validate_all(self):
+        if self._validate_general_parameters() and self._validate_tab(self.view.dimensions_tabs.current_index):
+            self._calculate_fields(self.view.dimensions_tabs.current_index)
+            return self._validate_vois() and self._validate_vois_cube()
+        return False
+
+    def _validate_general_parameters(self):
+        return self.view.hu_value.validate() and \
+               self.view.slice_offset.validate()
+
+    def _validate_tabs(self):
+        result = True
+        for index in range(len(self.view.dimensions_fields)):
+            if not self._validate_tab(index):
+                result = False
+        return result
+
+    def _validate_tab(self, index):
+        result = True
+        for field in self.view.dimensions_fields[index].values():
+            if not field.validate():
+                result = False
+        return result
+
+    def _validate_pixel_size(self):
+        dim = self.view.dimensions_fields[1]
+        fields = [dim["width"], dim["pixel_number_x"], dim["height"], dim["pixel_number_y"]]
+        if any(not field.text for field in fields):
+            return False
+
+        width = float(dim["width"].text)
+        pixel_number_x = int(dim["pixel_number_x"].text)
+        height = float(dim["height"].text)
+        pixel_number_y = int(dim["pixel_number_y"].text)
+        if math.isclose(width / pixel_number_x, height / pixel_number_y, abs_tol=1e-3):
+            for field in fields:
+                field.highlight_border(False)
+            return True
+        else:
+            for field in fields:
+                field.highlight_border(True)
+            return False
+
+    def _validate_vois(self):
+        result = True
+        voi_widgets = self.view.voi_scroll_area.widget().layout()
+        for index in range(voi_widgets.count() - 1):
+            voi_widget = voi_widgets.itemAt(index).widget()
+            if not voi_widget.validate():
+                result = False
```

Comment:
Why looping through all widgets ?
You could also call `return False` on the first one which is not valid and iterate further

Reply:
That would only highlight the first not-contained-widget.